src/zabapgit_repo_impl.prog.abap

```diff
@@ -533,7 +535,20 @@ CLASS lcl_repo IMPLEMENTATION.
 
     lt_cache = mt_local.
     lt_tadir = lcl_tadir=>read( get_package( ) ).
-    LOOP AT lt_tadir ASSIGNING <ls_tadir>.
+
+    IF it_filter[] IS INITIAL.
```

Comment:
Well, if the below sounds too nerdish, then just ignore and merge :)
But in my opinion this part really reduced code clarity. There is semantically strange `lt_tadir_aux` which is for the absolute majority of cases (I mean non-transport-save cases) just a copy of `lt_tadir`. And additional cycle. 
I would keep just one original loop + this code inside:

```abap

lt_filter = it_filter.
sort lt_filter by ...

...
LOOP AT lt_tadir ...
...

if lt_filter is not initial.
  read table lt_filter with key ... transporting no fields binary search.
  if sy-subrc > 0.
    continue.
  endif.
endif.
```

The code is actually shorter. The semantics is a lot clearer: _loop through records, if not in filter - skip_